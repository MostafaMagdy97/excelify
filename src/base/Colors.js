const baseColors = {
  primary: '#0070E0',
  secondary: '#06BEE1',
  success: '#44AF69',
  warning: '#FCAB10',
  danger: '#F8333C',
  black: '#1b2733',
  grey: '#637282',
  light: '#f5f5f5',
  white: '#fff',
  transparent: 'transparent',
};

const COLORS = {
  BLUE: 'blue',
  RED: 'red',
  DANGER: 'danger',
  PAGE_BACKGROUND: 'pageBackground',
  SUCCESS: 'success',
  WARNING: 'warning',
  LINK: 'link',
  DISABLED: 'disabled',
  TEXT: 'text',
  HELP_TEXT: 'helpText',
  BORDER: 'border',
  HEADER: 'header',
  WHITE: 'white',
  SEPARATOR: 'separator',
  TRANSPARENT: 'transparent',
};

const COLORS_VALUES = {
  [COLORS.BLUE]: baseColors.primary,
  [COLORS.PRIMARY_RED]: baseColors.secondary,
  [COLORS.DANGER]: baseColors.danger,
  [COLORS.PAGE_BACKGROUND]: baseColors.light,
  [COLORS.SUCCESS]: baseColors.success,
  [COLORS.WARNING]: baseColors.warning,
  [COLORS.LINK]: baseColors.primary,
  [COLORS.DISABLED]: baseColors.grey,
  [COLORS.TEXT]: baseColors.black,
  [COLORS.HELP_TEXT]: baseColors.grey,
  [COLORS.BORDER]: baseColors.light,
  [COLORS.HEADER]: baseColors.primary,
  [COLORS.WHITE]: baseColors.white,
  [COLORS.SEPARATOR]: baseColors.grey,
};

export { COLORS, COLORS_VALUES };
export default COLORS;
