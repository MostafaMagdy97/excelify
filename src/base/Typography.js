export const FONT_TYPES = {
  HERO_TITLE: 'heroTitle',
  SUPER_TITLE: 'superTitle',
  TITLE: 'title',
  HEADING: 'heading',
  SUBHEADING: 'subheading',
  BODY: 'body',
  CAPTION: 'caption',
};

export const FONT_SIZES = {
  // Breakpoints: ['400px', '576px', '786px', '992px']
  [FONT_TYPES.HERO_TITLE]: ['42px', '42px', '42x', '60px'],
  [FONT_TYPES.SUPER_TITLE]: ['32px', '32px', '32x', '30px'],
  [FONT_TYPES.TITLE]: ['21px', '21px', '20px', '20px'],
  [FONT_TYPES.HEADING]: ['19px', '19px', '18px', '18px'],
  [FONT_TYPES.SUBHEADING]: ['17px', '17px', '16px', '16px'],
  [FONT_TYPES.BODY]: ['15px', '15px', '14px', '14px'],
  [FONT_TYPES.CAPTION]: ['13px', '13px', '12px', '12px'],
};

export const FONT_WEIGHTS = {
  SEMI_BOLD: 600,
  NORMAL: 500,
  LIGHT: 400,
  SUPER_LIGHT: 300,
};

export const LINE_HEIGHTS = {
  [FONT_TYPES.HERO_TITLE]: '30px',
  [FONT_TYPES.SUPER_TITLE]: '30px',
  [FONT_TYPES.TITLE]: '24px',
  [FONT_TYPES.HEADING]: '24px',
  [FONT_TYPES.SUBHEADING]: '24px',
  [FONT_TYPES.BODY]: '24px',
  [FONT_TYPES.CAPTION]: '24px',
};
