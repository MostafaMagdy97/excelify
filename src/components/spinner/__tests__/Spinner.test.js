import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';

import { COLORS } from '../../../base/Colors';
import Spinner from '../Spinner';

configure({ adapter: new Adapter() });

describe('Spinner', () => {
  it('should render', () => {
    const component = <Spinner />;
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  it('should render radius correctly', () => {
    const component = <Spinner radius={60} />;
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  it('should render color correctly', () => {
    const component = <Spinner color={COLORS.PRIMARY_BLUE} />;
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });
});
