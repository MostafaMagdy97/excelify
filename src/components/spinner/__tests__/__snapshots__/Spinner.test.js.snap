// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render 1`] = `
.c0 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  width: 40px;
  height: 40px;
}

.c0 .path {
  stroke: #06BEE1;
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
}

<svg
  class="c0"
  radius="40"
  viewBox="0 0 50 50"
>
  <circle
    class="path"
    cx="25"
    cy="25"
    fill="none"
    r="20"
    stroke-width="4"
  >
    
  </circle>
</svg>
`;

exports[`Spinner should render color correctly 1`] = `
.c0 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  width: 40px;
  height: 40px;
}

.c0 .path {
  stroke: #06BEE1;
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
}

<svg
  class="c0"
  radius="40"
  viewBox="0 0 50 50"
>
  <circle
    class="path"
    cx="25"
    cy="25"
    fill="none"
    r="20"
    stroke-width="4"
  >
    
  </circle>
</svg>
`;

exports[`Spinner should render radius correctly 1`] = `
.c0 {
  -webkit-animation: cZxgpV 2s linear infinite;
  animation: cZxgpV 2s linear infinite;
  width: 60px;
  height: 60px;
}

.c0 .path {
  stroke: #06BEE1;
  stroke-linecap: round;
  -webkit-animation: kuSBtA 1.5s ease-in-out infinite;
  animation: kuSBtA 1.5s ease-in-out infinite;
}

<svg
  class="c0"
  radius="60"
  viewBox="0 0 50 50"
>
  <circle
    class="path"
    cx="25"
    cy="25"
    fill="none"
    r="20"
    stroke-width="4"
  >
    
  </circle>
</svg>
`;
